test/GovernorBravo.test.ts

```diff
@@ -0,0 +1,253 @@
+// Start - Support direct Mocha run & debug
+import 'hardhat'
+import '@nomiclabs/hardhat-ethers'
+// End - Support direct Mocha run & debug
+
+import chai, {expect} from 'chai'
+import {ethers} from 'hardhat'
+import {before} from 'mocha'
+import {solidity} from 'ethereum-waffle'
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers'
+import {advanceBlockTo} from './utils/index'
+import {
+    GovernorAlpha,
+    GovernorBravoDelegate,
+    GovernorBravoDelegator,
+    BitDAO,
+    Treasury,
+    Timelock
+} from '../typechain'
+
+// Wires up Waffle with Chai
+chai.use(solidity)
+
+const TREASURY_FUNDS = '1000000000000000000000000000'
+const VOTING_POWER = '100000000000000000000000000'
+
+const provider = ethers.provider
+const PROPOSAL_SPAN = 5
+const abiCoder = new ethers.utils.AbiCoder()
+// Governance.ProposalState in contracts/Governance.sol.
+enum ProposalState {
+    Pending,
+    Active,
+    Canceled,
+    Defeated,
+    Succeeded,
+    Queued,
+    Expired,
+    Executed
+}
+
+describe('GovernorBravoDelegate', function () {
+    before(async function () {
+        this.signers = await ethers.getSigners()
+        this.admin = this.signers[0]
+        this.delegatee1 = this.signers[1]
+        this.delegatee2 = this.signers[2]
+        this.delegatee3 = this.signers[3]
+        this.receiver = this.signers[4]
+        this.GovernorAlpha = await ethers.getContractFactory('GovernorAlphaV8')
+        this.GovernorBravoDelegate = await ethers.getContractFactory(
+            'GovernorBravoDelegate'
+        )
+        this.GovernorBravoDelegator = await ethers.getContractFactory(
+            'GovernorBravoDelegator'
+        )
+        this.Timelock = await ethers.getContractFactory('TimelockV8')
+        this.BitDAO = await ethers.getContractFactory('BitDAO')
+        this.Treasury = await ethers.getContractFactory('Treasury')
+        this.timelockAlpha = <Timelock>(
+            await this.Timelock.deploy(this.admin.address, 1)
+        )
+        await this.timelockAlpha.deployed()
+        this.timelock = <Timelock>(
+            await this.Timelock.deploy(this.admin.address, 1)
+        )
+        await this.timelock.deployed()
+        this.bit = <BitDAO>await this.BitDAO.deploy(this.admin.address)
+        await this.bit.deployed()
+        this.governorAlpha = <GovernorAlpha>(
```

Comment:
question: Why do you need GovAlpha in a GovBravo test?

Is GovBravo dependent on GovAlpha

Reply:
As per Compound GovernorBravo is derived from existing GovernorAlpha to make smooth upgrade from it to GovernorBravo as GovernorAlpha had no upgradeability itself and there is easy way to derive GovernorBravo (only need to pull current proposalCount as we didn't change GovernorAlpha state with those proposals except proposalCount).

Reply:
So to be clear, you're using the Alpha for the interface, to ensure Bravo can be interacted with using it?

Reply:
Alpha is need to init Bravo, its not used as in interface. Bravo is continuation from Alpha in literal sense.

Reply:
Thanks for pointing out this dependency @bitparadigm. I suggest we continue as is but then adapt Bravo to allow deploying without a prior instance of Alpha.

Reply:
@shahankhatch I wanted to first add GovenorBravo without updates to implementation except version upgrade to 0.8.x and very minor tweaks like comments and things like constants and `external` -> `public`. I can raise an issue to remove reliance on GovernorAlpha to init GovernorBravo and then make a PR specifically for it.